Source/RxComponents/LMCDampedHarmonicOscillator.hh

```diff
@@ -0,0 +1,52 @@
+/*
+ * LMCDampedHarmonicOscillator.hh
+ *
+ *  Created on: Jul 6, 2022
+ *      Author: pslocum
+ */
+
+#ifndef LMCEQUIVALENTCIRCUIT_HH_
+#define LMCEQUIVALENTCIRCUIT_HH_
+#include "LMCAnalyticResponseFunction.hh"
+#include "LMCConst.hh"
+#include "param.hh"
+#include "LMCException.hh"
+#include <math.h>
+
+namespace locust
+{
+    class DampedHarmonicOscillator : public AnalyticResponseFunction
+    {
+
+        public:
+    		DampedHarmonicOscillator();
+    		virtual ~DampedHarmonicOscillator();
+    		virtual bool Configure( const scarab::param_node& aNode );
+            virtual void GenerateFIR();
+            bool Initialize();
+            double GreensFunction(double t);
+            double ExpDecayTerm(double t);
+            std::vector<std::pair<double,double>> gfArray;
+
+
+        private:
+            double fCavityFrequency; // Hz
+            double fCavityOmega; // radians/s
+            double fCavityQ;
+            double fCavityDampingFactor;
+            double fBFactor; // harmonic oscillator parameter
+            double fCavityOmegaPrime;  // damped resonant frequency
+            int fMaxNBins;
```

Comment:
Is there a reason that we need an outside defined `fMaxNBins`? If not, I think that calculating the number of bins from the `fCavityQ` and `fTimeResolution` until the response has decayed below the `fThresholdFactor` would be more straigthforward.

Reply:
I was hoping to use `fMaxNBins` as an upper limit to avoid configuring something that takes longer than we anticipated.  Also, we have a debugging file that outputs the configured FIR in a text file for inspection, here:   https://github.com/project8/locust_mc/blob/master/Source/Core/LMCHFSSResponseFileHandler.cc#L135 .

Reply:
In principle we could exit the code with an error/warning if fMaxBins is required.

(Almost ready to test.)